herddb-core/src/main/java/herddb/codec/RecordSerializer.java

```diff
@@ -282,12 +304,13 @@ public static void skipTypeAndValue(ByteArrayCursor dii) throws IOException {
 
     public static byte[] serialize(Object v, int type) {
         if (v == null) {
-            return null;
+            throw new IllegalArgumentException("Object to be serialized cannot be null");
         }
         switch (type) {
             case ColumnTypes.BYTEARRAY:
                 return (byte[]) v;
             case ColumnTypes.INTEGER:
+            case ColumnTypes.NOTNULL_INTEGER:
```

Comment:
here you cannot serialize a NULL, for a NOTNULL_INTEGER, you have to duplicate all cases

Reply:
Done.